testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -203,6 +209,64 @@ impl BitcoinRegtestController {
         rest
     }
 
+    fn receive_blocks(&mut self, sync: bool) -> BurnchainTip {
+        if !self.use_coordinator {
+            return self.receive_blocks_helium();
+        }
+
+        let (mut burnchain, mut burnchain_indexer) = self.setup_indexer_runtime();
+        let (block_snapshot, state_transition) = loop {
+            match burnchain.sync_with_indexer(&mut burnchain_indexer) {
+                Ok(x) => {
+                    increment_btc_blocks_received_counter();
+                    // initialize the dbs...
+                    self.sortdb_mut();
```

Comment:
Shouldn't there be a separate method for db initialization?  And, shouldn't it be called before the loop begins?

Reply:
I believe it's set up this way because of a bootstrapping problem with getting the `first_burn_hash`, etc. from the indexer before it can initialize the databases.